src/device/stm32/stm32f40x-altfunc-bitfields.go

```diff
@@ -1,7 +1,7 @@
-// These are the supported alternate function numberings on the stm32f407
-// +build stm32,stm32f407
+// +build stm32
+// +build stm32f405 stm32f407
 
-// Alternate function settings on the stm32f4xx series
+// Alternate function numberings on the stm32f40x series
```

Comment:
This change should be separated into a separate PR.
The same goes for i2c.

If this is specific to stm32f4 you can add a buildtag like stm32f4.

Reply:
Before this PR, you might want to change your stm32f4disco configuration.
I'm not sure of the correct name, but the sources common to stm32f4disco and stm32f405 need to be cut out in a buildtag, for example stm32f4.

Then adding stm32f405 would work, I think.

Reply:
ah, `make smoketest` is what I needed to be running, not `make test`. thanks. 

I simply removed the pin definitions from `board_stm32f4disco.go`, since they can be shared by all boards with this chip, and now `make smoketest` passes as expected. 

are you suggesting i still make a new build tag anyway? there are several files that use that same 2-line build tag change I made like you are seeing there in `stm32f40x-altfunc-bitfields.go`. I should instead replace all instances of those 2 lines with the new build tag and update their target configurations?

Reply:
For this PR, it's better to change only the files that are needed for stm32f405.
It's easier for us to review.

> are you suggesting i still make a new build tag anyway?

yes.
An example of the change is as follows
https://github.com/sago35/tinygo/commit/1f74ac8089b629d23473fcc8986918a8da8c4112

Again, for this PR it's best to only make changes that pertain to stm32f405, so changes to the following files should be in a separate PR so they can be merged in first.

* src/machine/board_stm32.go
* src/machine/i2c.go
* targets/stm32f4disco.json
* src/device/stm32/stm32f407xx-altfunc-bitfields.go -> stm32f4-altfunc-bitfields.go
* src/machine/board_stm32f4disco.go
* src/machine/machine_stm32f407.go
* src/runtime/runtime_stm32f407.go

Reply:
thank you very much for the detailed feedback. hopefully I'm following your intent and will split this into the following new PRs:

- new build tag `stm32f4` - updates build constraints in existing board `stm32f4disco` and machine/runtime `stm32f407`
- new build tag `feather-stm32f405` - creates new target, utilizing new tag `stm32f4`, and implements board `feather-stm32f405` and machine/runtime `stm32f405`
- integrate UART support
- integrate SPI support
- integrate I2C support

please let me know if any of these are still too large.

(good thing I made this a draft!)

Reply:
I think it's fine to bundle everything in one PR for basic support (that way you can test the PR actually works!), but it would indeed be helpful to split out new peripheral support in separate PRs.